### PR TITLE
feature/AKD-126: Added Azure 'VM Desired SKU Size' plugin and test cases

### PR DESCRIPTION
**Pull request by ali-imran7**

### CODE REVIEW

File: plugins/azure/virtualmachines/desiredSkuSize.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'VM Desired SKU Size',
+    category: 'Virtual Machines',
+    description: 'Ensures that virtual machines is using the desired SKU size.',
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/desiredSkuSize.js:7]:
```suggestion
    description: 'Ensures that virtual machines are using the desired SKU size.',
```

File: plugins/azure/virtualmachines/desiredSkuSize.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var locations = helpers.locations(settings.govcloud);
+
+        const config = {
+            desiredSkuSize: settings.vm_desired_sku_size || this.settings.vm_desired_sku_size.default
+        };
+
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/desiredSkuSize.js:29]: `if (!config.desiredSkuSize.elngth) return callback(null, results, source);`

Comment (AkhtarAmir) [reply plugins/azure/virtualmachines/desiredSkuSize.js:29]: This should be an opt-in plugin

File: plugins/azure/virtualmachines/desiredSkuSize.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            if (!virtualMachines.data.length) {
+                helpers.addResult(results, 0, 'No existing virtual machines', location);
+                return rcb();
+            }
+
+            async.each(virtualMachines.data, function(virtualMachine, scb) {
+                let vmSkuSize = '';
+                if (virtualMachine.hardwareProfile && virtualMachine.hardwareProfile.vmSize) {
+                    vmSkuSize = virtualMachine.hardwareProfile.vmSize.toLowerCase();
+                }
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/desiredSkuSize.js:50]:
```suggestion
                } else return scb();
```

File: plugins/azure/virtualmachines/desiredSkuSize.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                return rcb();
+            }
+
+            if (!virtualMachines.data.length) {
+                helpers.addResult(results, 0, 'No existing virtual machines', location);
+                return rcb();
+            }
+
+            async.each(virtualMachines.data, function(virtualMachine, scb) {
+                let vmSkuSize = '';
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/desiredSkuSize.js:47]:
```suggestion
                let vmSkuSize;
```

File: plugins/azure/virtualmachines/desiredSkuSize.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    more_info: 'VM SKU size defines the compute power and data processing speed. VM SKU size should be chosen carefully to address compute requirements for the organization and to save un-necessary costs.',
+    recommended_action: 'Resize VM to desired SKU size.',
+    link: 'https://docs.microsoft.com/en-us/azure/virtual-machines/sizes',
+    apis: ['virtualMachines:listAll'],
+    settings: {
+        vm_desired_sku_size: {
+            name: 'VM Desired SKU Size',
+            description: 'SKU size for the virtual machines. Created virtual machine SKU sizes should match the desired SKU size',
+            regex: '/(^Basic_A(.*?)$)|(^Standard_(A|B|D|E|F|G|H|L|M|N)(.*?)$)/i',
+            default: 'standard_ds3_v2'
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/desiredSkuSize.js:17]:
```suggestion
            default: ''
```

File: plugins/azure/virtualmachines/desiredSkuSize.js (hunk 2 above)
Comment (AkhtarAmir) [reply plugins/azure/virtualmachines/desiredSkuSize.js:29]: Also make this setting a comma-separated list so that they can choose from a list

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/azure/virtualmachines/desiredSkuSize.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var locations = helpers.locations(settings.govcloud);
+
+        const config = {
+            desiredSkuSize: settings.vm_desired_sku_size || this.settings.vm_desired_sku_size.default
+        };
+
```
Comment (giorod3) [inline plugins/azure/virtualmachines/desiredSkuSize.js:29]: opt-in, should callback if not set

**Review by giorod3 — CHANGES_REQUESTED** [review]

**Comment by giorod3** [comment]
merge master as well

**Pull request merged**